common/tree/script_test.go

```diff
@@ -17,25 +18,294 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func fmtCode(t int) string {
+	return fmt.Sprintf("%x", t)
+}
+
+type ClosureTestRun struct {
+	run           string
+	script        string
+	expectedError bool
+}
+
 func TestDecodeClosure(t *testing.T) {
-	testCases := []struct {
-		script string
-	}{
+	var exPubKey1 = "f8352deebdf5658d95875d89656112b1dd150f176c702eea4f91a91527e48e26"
+	var exPubKey2 = "fc68d5ea9279cc9d2c57e6885e21bbaee9c3aec85089f1d6c705c017d321ea84"
+	var exPubKey3 = "fc68d5ea9279cc9d2c57e6885e21bbaee9c3aec85089f1d6c705c017d321ea84"
+	var sequenceExample = "00400007"
+	var disabledSequenceExample = "ffffffff"
+	var exampleLocktime = "001b5e67"
```

Comment:
let's make these constants instead of vars and move their declaration outside the testing function:

```
const (
	exPubKey1 = "f8352deebdf5658d95875d89656112b1dd150f176c702eea4f91a91527e48e26"
	exPubKey2 = "fc68d5ea9279cc9d2c57e6885e21bbaee9c3aec85089f1d6c705c017d321ea84"
	...
)

func TestDecodeClosure(t *testing.T) {
```